lib/assets/javascripts/cartodb/table/mapview.js

```diff
@@ -1318,14 +1318,6 @@ cdb.admin.MapTab = cdb.core.View.extend({
     this.mapView.$el.append(this.stackedLegend.render().$el);
     this.addView(this.stackedLegend);
     if (this.overlays) this.overlays.setLegend(this.stackedLegend);
-
-    if (this.legends && this.legends.length) {
```

Comment:
Adding this back I think should fix issues with the legend location inconsistency.